From 163ef815877db9a8057c8afce9867a82a6c8ee72 Mon Sep 17 00:00:00 2001
From: Miquel Raynal <miquel.raynal@bootlin.com>
Date: Thu, 18 Oct 2018 15:19:06 +0200
Subject: [PATCH 0724/1051] ARM64: dts: a3900-db-vd: Disable SPI-1 on board
 with enabled NAND

The NAND controller and the SPI bus on the CP share pins:
* MPP13 is NAND R/B# and SPI MISO
* MPP15 is NAND ADD7 and SPI MOSI
* MPP16 is NAND ADD6 and SPI CLK

This patch disables CP0 SPI-1 and adds a comment warning about
the impossibility to use both CP NAND and CP SPI-1 in the same HW
confiration.

Change-Id: I0bc11263b561e029a67c79b397ad3e23e220fb70
Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60389
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts | 10 +++++++++-
 arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts | 10 +++++++++-
 2 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts
index 65bdc42573c8..541bde9682c3 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts
@@ -66,8 +66,12 @@
 	status = "okay";
 };
 
+/*
+ * cp0_spi1 and cp0_nand_controller have common pins on this board. Hence,
+ * only one can be used at a time.
+ */
 &cp0_spi1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &cp0_sdhci0 {
@@ -78,6 +82,10 @@
 	status = "okay";
 };
 
+/*
+ * cp0_spi1 and cp0_nand_controller have common pins on this board. Hence,
+ * only one can be used at a time.
+ */
 &cp0_nand_controller {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts
index 669df36c08ef..f1e42fdffa6b 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts
@@ -67,8 +67,12 @@
 	status = "okay";
 };
 
+/*
+ * cp0_spi1 and cp0_nand_controller have common pins on this board. Hence,
+ * only one can be used at a time.
+ */
 &cp0_spi1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &cp0_sdhci0 {
@@ -79,6 +83,10 @@
 	status = "okay";
 };
 
+/*
+ * cp0_spi1 and cp0_nand_controller have common pins on this board. Hence,
+ * only one can be used at a time.
+ */
 &cp0_nand_controller {
 	status = "okay";
 };
-- 
2.17.1

